compiler/rustc_expand/src/expand.rs

```diff
@@ -529,9 +529,11 @@ impl<'a, 'b> MacroExpander<'a, 'b> {
     fn error_derive_forbidden_on_non_adt(&self, derives: &[Path], item: &Annotatable) {
         let attr = self.cx.sess.find_by_name(item.attrs(), sym::derive);
         let span = attr.map_or(item.span(), |attr| attr.span);
-        let mut err = self
-            .cx
-            .struct_span_err(span, "`derive` may only be applied to structs, enums and unions");
+        let mut err = self.cx.struct_span_err_with_code(
+            span,
+            "`derive` may only be applied to structs, enums and unions",
+            DiagnosticId::Error("E0774".into()),
+        );
```

Comment:
Maybe use `struct_span_err!` instead? That also makes sure `E0774` is a valid error code IRRC.